code/modules/mob/living/bot/bot.dm

```diff
@@ -435,11 +438,63 @@
 		if(D.dir == dir)		return 1
 
 	for(var/obj/machinery/door/D in loc)
+		if(loc.pathweight == 1)
+			loc.pathweight = 2	//Slight bias in pathfinding away from airlocks
```

Comment:
Err, if I'm reading this right, you're *updating* the pathweight in-place. We _probably_ shouldn't mess with the map variables or Weird Badness might happen.

Would be much cleaner to modify the Distance proc used - instead of reading the pathweight off of the turf directly, read it into a var, put in type-check for doors and add/multiply the base cost there.

Reply:
No excuses for this one, slipped by me. Meant to return to it later and completely forgot! I'll incorporate it into the distance proc